x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/ingest/SetSecurityUserProcessor.java

```diff
@@ -85,6 +86,39 @@ public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
                         userObject.put("metadata", user.metadata());
                     }
                     break;
+                case API_KEY:
+                    final HashMap<String, Object> apiKey = new HashMap<>();
```

Comment:
Per https://github.com/elastic/elasticsearch/pull/51454/, if `userObject` has an `api_key` field already, and it's an object (Map) then we should add new fields to that object, not overwrite it.
Likewise for realm.

Reply:
Updated. Also added test to make sure they are not overwritten.